Adding asyncio green mode documentation and a how-to on contributing

LJBD opened the pull request:
This PR contains documentation for asyncio green mode, a how-to on contributing and several other small changes.
It's still a WIP.
It references 1st item on the checklist in #110.

Commit:
Added PyCharm project files to gitignore.
pytango-110 PyTango workshop @ SOLARIS - list of tasks

Commit:
Added a small introduction about asyncio green mode, a paragraph for Client green modes and specified a Sphinx version in the Readme.
pytango-110 PyTango workshop @ SOLARIS - list of tasks

Commit:
Refactored green modes documentation and added links for issues and PRs in conf.py
pytango-110 PyTango workshop @ SOLARIS - list of tasks

Commit:
Refactored database from doc/tep to be compatible with Python 3.
pytango-110 PyTango workshop @ SOLARIS - list of tasks

Commit:
Added an example of an asyncio device and fixed importing GreenMode from tango.
pytango-110 PyTango workshop @ SOLARIS - list of tasks

lzytniak committed:
How co contribute documentation

lzytniak committed:
Merge branch 'pytango-110-docs' of github.com:tango-controls/pytango into pytango-110-docs
Content file was modified

AntoineDupre commented on README.rst:
```diff
@@ -39,7 +39,7 @@ Python dependencies:
 
 - setuptools_
-- sphinx_
+- sphinx_ < 1.5
```
This is not useful anymore, it has been fixed in PR #112.

AntoineDupre commented on doc/how-to-contribute.rst:
```diff
@@ -0,0 +1,46 @@
+A normal Git workflow is used. You can find how to automate your git branching workflow example_.
+Good practices:
+
+* You can create pull requests, you don't have to use your fork,
```
If someone is not a member, he needs to fork the project to contribute

Commit:
Merging develop

Commit:
Since Sphinx build is corrected for higher versions of Sphinx, this is not necessary anymore.
pytango-110 PyTango workshop @ SOLARIS - list of tasks

Commit:
Merge remote-tracking branch 'origin/pytango-110-docs' into pytango-110-docs

lzytniak committed:
Update after review

AntoineDupre commented on doc/how-to-contribute.rst:
```diff
@@ -0,0 +1,45 @@
+.. currentmodule:: tango
```
This should be added at the beginning of the how to section

Commit:
Added more information about server green modes.
pytango-110 PyTango workshop @ SOLARIS - list of tasks

Commit:
Merge remote-tracking branch 'origin/pytango-110-docs' into pytango-110-docs

AntoineDupre commented on doc/green_modes/green_modes_server.rst:
```diff
@@ -0,0 +1,39 @@
+coroutines, so you should define them with "async" keyword. DS. code written
+with asyncio. There's no monitor lock!
+
+.. literalinclude:: ../../examples/asyncio_device/asyncio_device_example.py
```
This path does not exist. 
Should it be ```../../examples/asyncio_green_mode/asyncio_device_example.py``` ?

LJBD replied:
Fixed.

Commit:
Added a short description of the asyncio device example.
pytango-110 PyTango workshop @ SOLARIS - list of tasks

lzytniak committed:
New section in HowTo, about how to contribute

Commit:
Removed unnecessary lines from asyncio device example.
pytango-110 PyTango workshop @ SOLARIS - list of tasks

lzytniak committed:
`How to Contribute` added

lzytniak committed:
Merge branch 'pytango-110-docs' of github.com:tango-controls/pytango into pytango-110-docs

MikeFalowski committed:
Add two examples for asyncio client and minor text fixes

MikeFalowski committed:
Divide client doc into files for each mode; add short description about TCP server example.

Clients examples for asyncio, gevent and futures are long, so now it is easier to edit them.

Explanation of TCP server example was needed to make its function more clear.
